azure/firewall/main.tf

```diff
@@ -0,0 +1,111 @@
+#############
+# Network
+#############
+
+resource "azurerm_public_ip" "main" {
+  name                = var.pip_name
+  resource_group_name = var.resource_group_name
+  location            = var.location
+  tags                = var.tags
+
+  allocation_method = "Static"
+  sku               = "Standard"
+}
+
+resource "azurerm_subnet" "main" {
+  name                 = "AzureFirewallSubnet"
+  resource_group_name  = var.resource_group_name
+  virtual_network_name = var.virtual_network_name
+  address_prefixes     = var.subnet_address_prefixes
+}
+
+#############
+# Firewall
+#############
+
+resource "azurerm_firewall" "main" {
+  name                = var.firewall_name
+  location            = var.location
+  resource_group_name = var.resource_group_name
+  sku_name            = var.firewall_sku_name
+  sku_tier            = var.firewall_sku_tier
+  dns_servers         = var.firewall_dns_servers
+  tags                = var.tags
+  firewall_policy_id  = var.firewall_policy_id
+
+  ip_configuration {
+    name                 = "configuration"
+    subnet_id            = azurerm_subnet.main.id
+    public_ip_address_id = azurerm_public_ip.main.id
+  }
+}
+
+#############
```

Comment:
delete code that's commented out. You can always go back to the previous commit if you need it in the future